lib/rules/no-missing-g-flag.ts

```diff
@@ -0,0 +1,151 @@
+import type { RegExpContext, UnparsableRegExpContext } from "../utils"
+import { createRule, defineRegexpVisitor } from "../utils"
+import type { ExpressionReference } from "../utils/ast-utils"
+import {
+    extractExpressionReferences,
+    isKnownMethodCall,
+} from "../utils/ast-utils"
+import { createTypeTracker } from "../utils/type-tracker"
+
+/**
+ * Parse option
+ */
+function parseOption(
+    userOption:
+        | {
+              strictTypes?: boolean
+          }
+        | undefined,
+) {
+    let strictTypes = true
+    if (userOption) {
+        if (userOption.strictTypes != null) {
+            strictTypes = userOption.strictTypes
+        }
+    }
+
+    return {
+        strictTypes,
+    }
+}
+
+export default createRule("no-missing-g-flag", {
+    meta: {
+        docs: {
+            description:
+                "disallow missing `g` flag in patterns used in `String#matchAll` and `String#replaceAll`",
+            category: "Possible Errors",
+            // TODO Switch to recommended in the major version.
+            // recommended: true,
```

Comment:
Why this not recommend?

Reply:
Because that would be a breaking change. Existing users that update the plugin might suddenly find their CI failing if we included it.

Reply:
Understood, thanks.